MergerLogic/DataTypes/Data.cs

```diff
@@ -278,14 +275,13 @@ public bool TileExists(Coord coord)
             }
             this._logger.LogDebug($"[{MethodBase.GetCurrentMethod().Name}] end for coord: {coords.ToString()}, upscale: {upscale}");
             fetchTileStopWatch.Stop();
-            this._metricsProvider.TotalFetchTimePerTileHistogram()?.Observe(fetchTileStopWatch.Elapsed.TotalMilliseconds);
+            this._metricsProvider.TotalFetchTimePerTileHistogram(fetchTileStopWatch.Elapsed.TotalMilliseconds);
             return correspondingTile;
         }
 
         public void UpdateTiles(IEnumerable<Tile> tiles)
         {
-            var batchUploadStopWatch = new Stopwatch();
-            batchUploadStopWatch.Start();
+            var batchUploadStopWatch = Stopwatch.StartNew();
```

Comment:
This variable only has this function as it's scope.
Please rename to something like `stopwatch`.

Reply:
done